src/Authentication/Authenticators/Session.php

```diff
@@ -646,8 +646,8 @@ public function startLogin(User $user): void
         /** @var int|string|null $userId */
         $userId = $this->getSessionUserKey('id');
 
-        // Check if already logged in.
-        if ($userId !== null) {
+        // Check if already logged in and magic link force login config.
+        if ($userId !== null && config('Auth')->allowMagicLinkForceLogin === false) {
```

Comment:
This seems to invalidate the following security check.
So I think this is a security hole.

Reply:
do we have to delete the old session then pass the condition?